source/validate_id.cpp

```diff
@@ -188,14 +188,18 @@ bool idUsage::isValid<SpvOpEntryPoint>(const spv_instruction_t* inst,
                           << "' is not a function.";
     return false;
   }
-  // TODO: Check the entry point signature is void main(void), may be subject
-  // to change
-  auto entryPointType = usedefs_.FindDef(entryPoint.second.words[4]);
-  if (!entryPointType.first || 3 != entryPointType.second.words.size()) {
-    DIAG(entryPointIndex) << "OpEntryPoint Entry Point <id> '"
-                          << inst->words[entryPointIndex]
-                          << "'s function parameter count is not zero.";
-    return false;
+  // don't check kernel function signatures
```

Comment:
Actually, I don't see why this check exists at all, even for graphics entry points.  I can't find this restriction anywhere in the SPIR-V spec.  @umar456 do you happen to know why an entry point function with parameters would be considered invalid?